### ISSUE TITLE
[CLOSED] Development build type that causes Control Center to trace loops

### ISSUE THREAD

**Issue by BlueFinBima** [BlueFinBima/test1#176]
<a href="https://github.com/derammo"><img src="https://avatars0.githubusercontent.com/u/817160?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [derammo](https://github.com/derammo)**
_Friday Jan 10, 2020 at 15:12 GMT_
_Originally opened as https://github.com/BlueFinBima/Helios/pull/176_

----

Soft loops are binding loops that are not circular, but probably are still an error, because they come from some interface or visual and return back to the same one.  These are usually errors created when binding network interfaces.

This PR adds a new Configuration to the build, called "Development".  It is not intended for release at this time, since it has no UI.   It builds to the same output as "Debug" (..\bin\Debug) and overwrites Helios.dll and Control Center.exe with the Development versions.   This is to minimize impact on all the other projects that copy their output to ..\bin\Debug and ..\bin\Release.   If this is a problem, there is some messy work to add another output directory.  

When running the Development build, conditional code is added to Helios bindings and Helios objects to detect bindings that are not technically circular but which return to the same object.  These are logged as Info, because they are not necessarily a mistake   Output looks like this:

1/10/2020 9:53:30 AM - Info - Soft binding loop detected; Object DCS A10C may have triggered itself
1/10/2020 9:53:30 AM - Info -   binding loop includes when value on UHF rotator changes set Intercom UHF Volume on DCS A10C to  trigger value
1/10/2020 9:53:30 AM - Info -   binding loop includes when Intercom UHF Volume on DCS A10C changes set value on UHF rotator to  trigger value
1/10/2020 9:53:30 AM - Info -   binding loop trace complete

All of these are probable mistakes in not setting cascading trigger bypass.  This isn't a significant enough problem to include this code in the field I think, but I wanted to be able to make a dev build that we can use to find these sort of things.  If nobody is worried about them, then please reject this pull request.

The implementation is as small as I could make it, even using statics to simplify the logic.  The statics work correctly because all triggering is done on the main thread in Helios it seems. 

----

_**[derammo](https://github.com/derammo)** included the following code: https://github.com/BlueFinBima/Helios/pull/176/commits_

**Comment by BlueFinBima** [comment]
<a href="https://github.com/BlueFinBima"><img src="https://avatars0.githubusercontent.com/u/18526232?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [BlueFinBima](https://github.com/BlueFinBima)**
_Monday Feb 17, 2020 at 08:47 GMT_

----

This needs to be documented so that others performing development on interfaces / components can identify these problems.

**Comment by BlueFinBima** [comment]
<a href="https://github.com/derammo"><img src="https://avatars0.githubusercontent.com/u/817160?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [derammo](https://github.com/derammo)**
_Monday Feb 17, 2020 at 12:47 GMT_

----

ultimately this is only useful for profile developers like CZ because all these loops are from cascading trigger not being bypassed.  So CZ can compile the product himself but if we really think this is useful then we should develop (carefully) a command line switch to turn it on.   I made it conditional compile because it is so expensive and touches core things, but we could carefully make it a static switch if we can make it not expensive when turned off.

Related:  I think we discussed making it not cascade by default (profile editor change) because you pretty much never want that. 

should check if that change happened
